### PR TITLE
[Build] missing torchsnapshot dependency

### PR DESCRIPTION
**Pull request by apbard**

### CODE REVIEW

**Review by vmoens — APPROVED** [review]
~~LGTM~~ see my comment below

**Review by vmoens — APPROVED** [review]
LGTM thanks!

File: test/test_tensorclass_nofuture.py [hunk 1]
```diff
@@ -11,7 +11,15 @@
...
 import torch
-import torchsnapshot
+
+try:
+    import torchsnapshot
+
+    _has_torchsnapshot = True
+except ImportError as err:
+    _has_torchsnapshot = False
+    TORCHSNAPSHOT_ERR = str(err)
```
Comment (vmoens) [inline test/test_tensorclass_nofuture.py:21]: this should be created out of the try/except to be visible
TORCHSNAPSHOT_ERR=None will suffice

Comment (apbard) [reply test/test_tensorclass_nofuture.py:21]: done in the try...except block like we did for TORCHREC

**Review by vmoens — APPROVED** [review]

**Pull request merged**